### ISSUE TITLE
Tooltips not appearing on hover when scrolling down a React based website

### ISSUE THREAD

**Issue by neekfenwick** [DevExpress/testcafe#3564]
### What is your Test Scenario?
Iterating reviews on a site, some data is only presented in tooltips, dynamically created HTML elements.  We must hover over an element in each row to allow the popup to appear, then inspect the popup's nodes for data.

*Please note* this test iterates useless 'Notify' popups that are all the same.   We are actually having the same bug but on a page only accessible once you log into the business side of the portal.  Solving the bug presented here should also fix the bug behind the login gate.
 
### What is the Current behavior?
The page scrolls correctly to the node being hovered, but after one or two popups no further popups appear.
 
### What is the Expected behavior?
Every node that is hovered should produce its correct popup tooltip.
 
### What is your web application and your TestCafe test code?

 
Your website URL (or attach your complete example):
It is not our website.  I haven't been able to recreate the problem myself.  They use React but I cannot tell what library or tool they are using for their popups, it doesn't seem to be react-tooltip for example.  So, I cannot provide a simple test case, only reproducing the error on this third party site.  Sorry :)

<details>
<summary>Your complete test code (or attach your test files):</summary>

 
```js
import { Selector } from 'testcafe';

fixture('Testing Begin...')
    .page('https://www.trustpilot.com/review/www.lego.com');

test('List names', async t => {

    let reviewRow = Selector('div.review-card');

    let numReviews = await reviewRow.count;
    console.log(`Iterating ${numReviews} rows...`);

    for (let idx = 0 ; idx < numReviews ; idx ++) {
        let row = reviewRow.nth(idx);

        let nameNode = row.find('div.consumer-information__name');
        let notifyNode = row.find('button.has-tooltip');
        let notifyNodeCount = await notifyNode.count;
        let name = await nameNode.innerText;
        console.log(`Going to hover name ${name} with ${notifyNodeCount} nodes to hover...`);

        await t
            .hover(notifyNode)
            .wait(500)
            .takeScreenshot('row-' + idx + '-' + name.replace(/[ \.]/, '-').trim() + '.png');
        console.log(`Done for ${idx} with name ${name}`);
    }
    console.log('Finished.');
});
```
</details>

<details>
<summary>Your complete test report:</summary>

``` Testing Begin...
 Running tests in:
 - Chrome 66.0.3359 / Linux 0.0.0

 Testing Begin...
Iterating 20 rows...
Going to hover name Julien Santoni with 1 nodes to hover...
Done for 0 with name Julien Santoni
Going to hover name madison butler with 1 nodes to hover...
Done for 1 with name madison butler
Going to hover name Jason Coller with 1 nodes to hover...
Done for 2 with name Jason Coller
Going to hover name P-A-C with 1 nodes to hover...
Done for 3 with name P-A-C
Going to hover name Jason Shepherd with 1 nodes to hover...
Done for 4 with name Jason Shepherd
Going to hover name Joseph Fidalgo with 1 nodes to hover...
Done for 5 with name Joseph Fidalgo
Going to hover name customer with 1 nodes to hover...
Done for 6 with name customer
Going to hover name D J Jones with 1 nodes to hover...
Done for 7 with name D J Jones
Going to hover name Guy Freedman with 1 nodes to hover...
Done for 8 with name Guy Freedman
Going to hover name nate with 1 nodes to hover...
Done for 9 with name nate
Going to hover name Alistair with 1 nodes to hover...
Done for 10 with name Alistair
Going to hover name Mark with 1 nodes to hover...
Done for 11 with name Mark
Going to hover name Mr L. with 1 nodes to hover...
Done for 12 with name Mr L.
Going to hover name Matt Barty with 1 nodes to hover...
Done for 13 with name Matt Barty
Going to hover name Christopher with 1 nodes to hover...
Done for 14 with name Christopher
Going to hover name Carlitos Gneiting with 1 nodes to hover...
Done for 15 with name Carlitos Gneiting
Going to hover name Jay with 1 nodes to hover...
Done for 16 with name Jay
Going to hover name Louise mccombes with 1 nodes to hover...
Done for 17 with name Louise mccombes
Going to hover name Jack Kvatch with 1 nodes to hover...
Done for 18 with name Jack Kvatch
Going to hover name Peter D with 1 nodes to hover...
Done for 19 with name Peter D
Finished.



```
</details>

<details>
<summary>Screenshots:</summary>

After the first few screenshots, in which the tooltip popup appears OK, no further popups appear, I will just provide the first few, you should see the error, after two that work the last tooltip gets stuck on screen and scrolled up without any new tooltips appearing.

Headless seems to behave slightly worse than non-headless but still goes wrong.

```
![row-3-P-A-C](https://user-images.githubusercontent.com/266415/54331103-fd238b00-464a-11e9-87ba-1318138e98f8.png)
![row-2-Jason-Coller](https://user-images.githubusercontent.com/266415/54330680-3ce97300-4649-11e9-8a99-7bbf0ac7abcf.png)
![row-1-madison-butler](https://user-images.githubusercontent.com/266415/54330681-3d820980-4649-11e9-918f-fe757c2b444c.png)
![row-0-Julien-Santoni](https://user-images.githubusercontent.com/266415/54330682-3d820980-4649-11e9-8b16-9619b316fc96.png)


```
</details>

### Steps to Reproduce:

1. Run the testcafe script given above in headless mode with something like:
`./node_modules/.bin/testcafe chrome:headless bug.ts -S -s screenshots --selector-timeout 60000 --assertionimeout 60000 --screenshot-on-fail --skip-js-errors`

2. Screenshots will be created in `screenshots` directory numbered and with the name of the reviewer for reference.

3. Run the same script in non-headless mode, for comparison to show it also suffers the bug.

4. Screenshots in `screenshots` directory as before or just watch the browser run the test.
 
### Your Environment details:
 
* testcafe version:                   1.1.0
* node.js version:                    v9.11.1
* command-line arguments:  ./node_modules/.bin/testcafe chrome:headless bug.ts -S -s screenshots --selector-timeout 60000 --assertionimeout 60000 --screenshot-on-fail --skip-js-errors
* browser name and version: Version 66.0.3359.117 (Official Build) (64-bit)
* platform and version:          Fedora Linux 29
* other:                                   The page does seem to use infinite scroll techniques to load data as you scroll, but the set of rows in which the error occurs is definitely loaded when the test begins.  I explicitly count existing rows before iterating, so I only expect to process the number of rows that exist when the test begins, and not try to continue down as more rows are loaded while scrolling.

**Comment by miherlosev** [comment]
Hi @neekfenwick 

I've reproduced the problem. Thank you for the provided detail.

**Comment by miherlosev** [comment]
There is a quick workaround. 
Need to insert `click` action before `hover` action.
```js
import { Selector } from 'testcafe';

fixture('Testing Begin...')
    .page('https://www.trustpilot.com/review/www.lego.com');

test('List names', async t => {

    let reviewRow = Selector('div.review-card');

    let numReviews = await reviewRow.count;
    console.log(`Iterating ${numReviews} rows...`);

    for (let idx = 0 ; idx < numReviews ; idx ++) {
        let row = reviewRow.nth(idx);

        let nameNode = row.find('div.consumer-information__name');
        let notifyNode = row.find('button.has-tooltip');
        let reviewActions = row.find('.review__actions');
        let notifyNodeCount = await notifyNode.count;
        let name = await nameNode.innerText;
        console.log(`Going to hover name ${name} with ${notifyNodeCount} nodes to hover...`);

        await t
            .click(reviewActions)
            .hover(notifyNode)
            .wait(500)
            .takeScreenshot('row-' + idx + '-' + name.replace(/[ \.]/, '-').trim() + '.png');
        console.log(`Done for ${idx} with name ${name}`);
    }
    console.log('Finished.');
});

```

**Comment by neekfenwick** [comment]
I see the site design at the URL I used in the test has changed so that popups do not appear in the same way any more :( This may mean we cannot test the fix in the same environment as it was reported.

I have tried rewriting the test script to use the "Useful" popups, where at least one person has marked a review as Useful then a popup does appear there, but they seem to be using a different mechanism as before, and hovering the Useful node does not cause the popup to appear.

If/When a fix does become available I'm happy to test it against the real site, behind the login gate, if you advise me how to get hold of the fixed version.
